### PR TITLE
Performance Improvements

### PR DESCRIPTION
**Pull request by rbock44**
- added a Makefile to live/production to simplify setup of terraform
- added httpd config to mange configure apache from terraform
- added php.ini to use production settings instead of development
- set limits and requests to same value to get Guaranted QoS

### CODE REVIEW

File: modules/gcloud/variables.tf [hunk 1]
```diff
@@ -33,3 +33,7 @@ variable "service_account" {
 variable "machine_type" {
   description = "The name of a Google Compute Engine machine type."
 }
+
+variable "issue_client_certificate" {
+  description = "Create client certificates which finally means create a new cluster"
```
Comment (marinusma) [inline modules/gcloud/variables.tf]: "Whether client certificate authorization is enabled for this cluster."

File: live/production/Makefile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# apply terraform with secrets
+.PHONY: apply
+apply:
+	# just make sure we know what we are doing
+	terraform apply -var-file secrets/terraform-production.tfvars
+
+.PHONY: create
+create:
+	# just make sure we know what we are doing
+	terraform apply -var-file secrets/terraform-production.tfvars var="issue_client_certificate=true"
```
Comment (marinusma) [inline live/production/Makefile]: Shouldn't be a variable. Please use default issue_client_certificate=true

File: live/production/Makefile [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+PLATFORM ?= linux_amd64
+HOME_PLUGINS_DIR=$(HOME)/.terraform.d/plugins/$(PLATFORM)
+LOCAL_PLUGINS_DIR=.terraform/plugins/$(PLATFORM)
+CUSTOM_PLUGIN=$(PLUGINS_DIR)/terraform-provider-kubernetes_v1.3.0-custom_x4
+CUSTOM_PLUGIN_DOWNLOAD_URL=https://github.com/sl1pm4t/terraform-provider-kubernetes/releases/download/v1.3.0-custom/terraform-provider-kubernetes_v1.3.0-custom_$(PLATFORM).zip
+
+# install custom kubernetes plugin to support ingress resource
+.PHONY: patch
+patch:
```
[the remark this answers is not in the document]
Comment (rbock44) [reply live/production/Makefile]: OK I removed patch and the UNAME magic as we no longer require it.
I kept the sql cloud proxy goal but added a TODO to read the name from ENV or figure out some way to automatically retrieve it.

File: live/production/Makefile (hunk 2 above)
Comment (rbock44) [reply live/production/Makefile]: I removed create.

File: live/production/variables.tf [hunk 4]
```diff
@@ -5,6 +5,11 @@ variable "cluster_password" {
   description = "Password for the kubernetes-cluster"
 }
 
+variable "issue_client_certificate" {
+  description = "Create new certificates which will basically create a new cluster when set to true"
+  default = false
```
[the remark this answers is not in the document]
Comment (rbock44) [reply live/production/variables.tf]: still wants to drop the cluster. I keep it to false for the moment.

File: modules/athene2/main.tf [hunk 5]
```diff
@@ -72,15 +72,21 @@ resource "kubernetes_deployment" "athene2_deployment" {
 
           resources {
             limits {
-              cpu    = "0.4"
-              memory = "500Mi"
+              cpu    = "0.1"
```
[the remark this answers is not in the document]
Comment (rbock44) [reply modules/athene2/main.tf]: yes we fix this when we are on latest

File: modules/athene2/main.tf [hunk 6]
```diff
@@ -72,15 +72,21 @@ resource "kubernetes_deployment" "athene2_deployment" {
...
+              memory = "200Mi"
             }
 
             requests {
-              cpu    = "250m"
+              cpu    = "0.1"
```
[the remark this answers is not in the document]
Comment (rbock44) [reply modules/athene2/main.tf]: yes we fix this when we are on latest

File: modules/athene2/main.tf [hunk 7]
```diff
@@ -234,3 +271,33 @@ resource "google_sql_database" "users" {
   charset   = "utf8"
   collation = "utf8_general_ci"
 }
+
+resource "kubernetes_config_map" "athene2_httpd_conf" {
```
[the remark this answers is not in the document]
Comment (rbock44) [reply modules/athene2/main.tf]: ok put them to athene2_conf config map which will be shared for all container in the pod

File: modules/gcloud/variables.tf (hunk 1 above)
Comment (rbock44) [reply modules/gcloud/variables.tf]: fixed comment

**Comment by marinusma** [comment]
Will be fixed/merged in https://github.com/serlo/infrastructure/pull/16

**Pull request closed without merging**